CLDC-4074 Update tests now that 2023 is closed

davidm-m opened the pull request:
Not finished, as I assume more tests will fail now. This is the first one I noticed and I saw we could fix it in a way that would remain fixed even in future years.

davidm-m committed:
CLDC-4074 Updated sales log tests to use filters from the current year instead of a fixed one

samyou-softwire committed:
CLDC-4074: Update other sales log tests to use the current year instead of a fixed one

samyou-softwire committed:
CLDC-4074: Update more failing tests

samyou-softwire committed:
CLDC-4074: Remove some 2024 sales log tests

samyou-softwire committed:
CLDC-4074: Ignore a failing request test

samyou-softwire committed:
CLDC-4074: Fix lettings log services tests

samyou-softwire committed:
CLDC-4074: Fix merge request helper spec

samyou-softwire committed:
CLDC-4074: Fix rake tests for 2024 logs

samyou-softwire committed:
CLDC-4074: Fix guidance helper tests

davidm-m reviewed (commented):
Really good stuff here, I like the forward-thinking approach to a bunch of these by not having fixed dates.
We should think about how to prevent this surprising us in future years. Can we update the documentation for developing forms for a year to include testing that? One way we can do it is by including some code in `config/environments/test.rb` that sets the time the tests think it is:
```ruby
config.after_initialize do
  new_forms_date = Time.local(2026, 4, 1, 10, 0, 0)
  old_forms_close_date = Time.local(2026, 7, 1, 10, 0, 0)
  Timecop.travel(new_forms_date)
end
```
We'd want it commented out but there so the devs working on the forms can use it easily. Any thoughts?

samyou-softwire committed:
CLDC-4074: Fix skipped tests

samyou-softwire commented:
> Really good stuff here, I like the forward-thinking approach to a bunch of these by not having fixed dates. We should think about how to prevent this surprising us in future years. Can we update the documentation for developing forms for a year to include testing that? One way we can do it is by including some code in `config/environments/test.rb` that sets the time the tests think it is:
> 
> ```ruby
> config.after_initialize do
>   new_forms_date = Time.local(2026, 4, 1, 10, 0, 0)
>   old_forms_close_date = Time.local(2026, 7, 1, 10, 0, 0)
>   Timecop.travel(new_forms_date)
> end
> ```
> 
> We'd want it commented out but there so the devs working on the forms can use it easily. Any thoughts?

looks like this is [already documented](https://mhclgdigital.atlassian.net/wiki/spaces/MC/pages/477593604/Check+the+automated+tests+will+pass+after+1st+April)

samyou-softwire committed:
CLDC-4074: Update tests for 2023 edit closing (#3095)

* CLDC-4074: Fix model tests

* CLDC-4074: Fix requests tests

* CLDC-4074: Fix services tests

* CLDC-4074: Fix rake tasks tests

the broken ones only make sense for 2024 logs so feels like timecop is the only option

* CLDC-4074: Specify why some tests are not validated based on year
